python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -261,6 +261,20 @@ def partition_key(self) -> str:
         """The partition key for the current run.
 
         Raises an error if the current run is not a partitioned run.
+
+        Examples:
+            .. code-block:: python
+
+                partitions_def = DailyPartitionsDefinition("2023-08-20")
+
+                @asset(
+                    partitions_def=partitions_def
+                )
+                def my_asset(context: AssetExecutionContext):
+                    context.log.info(context.partition_key)
+
+                # materializing the 2023-08-21 partition of this asset will log:
+                #   "2023-08-21"
```

Comment:
We should also note that this raises if the run is operating over a range, rather than a partition key. @petehunt ran into this recently.